### PR TITLE
Fixes for custom promise early failures

### PR DESCRIPTION
**Pull request by vpodzime**
Fixes CFE-3651 and CFE-3645.

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6624/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6624/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6624/

File: libpromises/mod_custom.c [hunk 1]
```diff
@@ -1034,6 +1037,8 @@ PromiseResult EvaluateCustomPromise(EvalContext *ctx, const Promise *pp)
     if (!success)
     {
         assert(interpreter == NULL && path == NULL);
+        /* Details logged in GEtInterpreterAndPath() */
```
Comment (larsewi) [inline libpromises/mod_custom.c:1040]: Possible typo `GEtInterpreterAndPath()` should be `GetInterpreterAndPath()` ?

**Review by larsewi — CHANGES_REQUESTED** [review]
Just one nitpick

**Review by larsewi — APPROVED** [review]
ACK

**Review by olehermanse — APPROVED** [review]
Great!

**Pull request merged**